client/src/types/User.ts

```diff
@@ -0,0 +1,24 @@
+export type CurrentUser =
+  | {
+      id: string;
+      name: string;
+      profilePicture: string;
+      chatRooms: string[];
+    }
+  | undefined;
+
+export type User = {
```

Comment:
Why 2 different types. CurrentUser and User. There can be only one type which is user